### DISCUSSION TITLE
Lesson 11 NFT: Mapping confusion & IPFS unable to load

### DISCUSSION THREAD

**Discussion by aloyboi** [smartcontractkit/full-blockchain-solidity-course-py#1430] [Q&A] [answered]
I am currently on the Lesson for NFTs and there is something that I cannot seem to grasp. I feel like it is a simple concept and yet I still cannot really comprehend

In the AdvancedCollectible.sol contract the tokenIdToBreed mapping maps the iterative tokenId(0-2) to a randomly selected breed 
![image](https://user-images.githubusercontent.com/78015506/164360365-da9db2be-e70d-4293-a3bb-727d30eef4b0.png)

In the create_metadata.py script, there is this line 'advanced_collectibles.tokenIdToBreed(token_id)'. What exactly does the advanced_collectibles.tokenIdToBreed(token_id) do? How is it able to obtain information from the enum BREED (advancedCollectible contract) and knows that the token_id maps to that specific breed in the enum BREED?

i.e if token_id = 0 was mapped to breed SHIBA_INU
advanced_collectibles.tokenIdToBreed(token_id) = 1 
![image](https://user-images.githubusercontent.com/78015506/164360484-8afa0776-fad5-4528-ae53-76b525be058f.png)

Also, I cant seem to load the metadata link on IPFS even while I am running the IPFS node, receiving the 505 gateway error.

Appreciate the help I can get!

**Comment by aloyboi** [comment]
In addition to my questions, when I am creating collectibles, I actually have a repeated breed for token_id 0 and 1, does the course tutorial tackle this problem?

**Comment by cromewar** [comment, marked as answer]
Hello @aloyboi 
Interesting question, let's try to answer it!
`tokenIdToBreed` is mapping (similar to a dict on python) so it relates a number with a Breed object (enum), but python does not understand the logic behind Enums on Solidity, so this way we can pass a simple integer into the python script and brownie will just run it, internally that integer will reference the Enum object containing the breeds of each PUG. And remember that a Enum is just a nice way to enumerate stuff, like 0 BREED1, 1 BREED2 and so on.

> In addition to my questions, when I am creating collectibles, I actually have a repeated breed for token_id 0 and 1, does the course tutorial tackle this problem?

You mean you repeated it intentionally or the code itself does not handle that? In any case the BREED is not a unique identifier, so you can have a lot of pugs with the same BREED, as this is a simple project the code does not go deeper, and create generative NFT's with different layers.

> Also, I cant seem to load the metadata link on IPFS even while I am running the IPFS node, receiving the 505 gateway error
This is harder to solve, as this could be for many reasons, does the problem persists? 

This is harder to answer as this could be for many reasons, like a firewall configuration, port conflicts, bad gateway and so on, so I'll need more info on this. Are you running IPFS desktop app or CLI?

**Reply by aloyboi** [reply]
Thank you for your answers. 

'Also, I cant seem to load the metadata link on IPFS even while I am running the IPFS node, receiving the 505 gateway error
This is harder to solve, as this could be for many reasons, does the problem persists?'

Did you answer this part? I saw you tag it but couldnt find a reply

**Reply by cromewar** [reply]
Hello @aloyboi sorry for the delay, was you able to change the network server to select the correct one?
